### ISSUE TITLE
Certain queries fail due to parser issue

### ISSUE THREAD

**Issue by Fhffsthu** [zino-app/graphql-flutter#264]
There is an issue in the graphql_parser with certain queries - see [https://github.com/angel-dart/graphql/issues/23](https://github.com/angel-dart/graphql/issues/23) for the issue and sample queries. 

The parsing failure causes the "operationName" to be set to null. In the 1.0.0+4 (as opposed to 1.0.0-beta.1+1) this has been detected and fixed by changing a null "operationName" to `'UNNAMED/' + document.hashCode.toString();`. This seems to guarantee the query will fail. Commented out this line seems to allow the queries to work - though I am guessing it will break something else.
 
This query will fail:
```
query customerMemberAttributes($memberId: Int!){
  customerByCustomerId(customerId: 7) {
    customerMemberAttributesByCustomerId {
      nodes {
        customerMemberAttributeId
        memberAttributesByCustomerMemberAttributeId(condition: {memberId: $memberId}) {
          nodes {
            memberAttributeId
          }
        }
      }
    }
  }
}
```
this query does not fail:
```
query customerMemberAttributes($memberId: Int!){
  customerByCustomerId(customerId: 7) {
    customerMemberAttributesByCustomerId {
      nodes {
        customerMemberAttributeId
        memberAttributesByCustomerMemberAttributeId(condition: {memberId: 7}) {
          nodes {
            memberAttributeId
          }
        }
      }
    }
  }
}
```

I am unsure what issue the check for the null query name fixed but the current version of the library does fail a few of my queries.

**Comment by endigo** [comment]
@HofmannZ Related to #235 

I have the same problem.

**Comment by micimize** [comment]
(hopeflly) closed by #276
